packages/api-fetch/src/index.js

```diff
@@ -119,6 +119,66 @@ function apiFetch( options ) {
 
 apiFetch.use = registerMiddleware;
 
+const capPerPageQuery = ( options ) => ( {
+	...options,
+	// Swap back to requesting the max of 100 items per page.
+	// TODO: This feels brittle. Is there a better way to manage request parameters?
+	path: options.path && `${ options.path.replace( /(\&?per_page)=-1/, '$1=100' ) }&page=1`,
+	url: options.url && `${ options.url.replace( /(\&?per_page)=-1/, '$1=100' ) }&page=1`,
```

Comment:
Can you use `@wordpress/url`'s [`addQueryArgs`](https://github.com/WordPress/gutenberg/blob/master/packages/url/src/index.js#L21-L35)?

Reply:
what about just setting a flag on the `options` for turning on paging?

Reply:
`per_page` seems to be too specific to WP queries.  This package is meant to be wp-agnostic I think?   Maybe `options` could accept something like:
```
{
     page: 'page'
}
```

Where the presence of that key "turns on" paging and the value is the paging arg that gets appended to the request url?

Reply:
I was not aware this package was intended to be wp-agnostic; no solution we come up with will be guaranteed to work with an arbitrary API. We can use another option here, but we will have to be making assumptions about how the WordPress REST API works in order for this to address our specific issue.

Reply:
It was my impression that published packages were intended to be wp-agnostic but I'll leave for others to confirm.  If so, maybe the middleware itself would be a part of wp-core then?

Reply:
nvm my points, I see that apiFetch itself is wp-agnostic but the bundled middlewares don't have to be.

Reply:
That said, that does represent an argument in favor of having WP apply this middleware rather than having it applied by default. I'll take that into consideration :+1: